module/surveys/helper.js

```diff
@@ -1036,6 +1060,30 @@ module.exports = class SurveysHelper {
                     submissionDocumentEvidences = surveySubmissionDocument[0].evidences;
 
                 } else {
+                    // join survey's program. PII data consent is given via this api call.
+                    if ( solutionDocument.programId && userToken !== "" ) {
+                        // not checking if already joined the program. it is handled in ml-core joinProgram fn.
+                        
+                        if ( programDocument.length > 0 ) {
```

Comment:
@VISHNUDAS-tunerlabs if this same thing is used across survey and observation we can have a common function